test/unit/spec/MfaVerify_spec.js

```diff
@@ -170,15 +170,36 @@ function (Q,
       ]);
     }
 
+    function mockChrome(isAvailable){
+      BrowserFeatures.isChrome = function () {};
+      spyOn(BrowserFeatures, 'isChrome').and.returnValue(isAvailable);
+    }
+
+    function mockFirefox(isAvailable){
+      BrowserFeatures.isFirefox = function () {};
+      spyOn(BrowserFeatures, 'isFirefox').and.returnValue(isAvailable);
+    }
 
     function setupU2F(options) {
-      if (!window.u2f || !window.u2f.sign) {
-        window.u2f = { sign: function () {} };
+      options || (options = {});
+
+      mockChrome(!options.noChrome);
```

Comment:
We have more tests with `hasU2f:true` than tests with `hasU2f:false`, so by default, it is better to have `noU2f:false` to avoid duplication of `hasU2f:true`